### PR TITLE
GH-924: Documented watch mode for tests.

### LINKED ISSUE #924: Document how to run tests in watch mode.
I could do something like this on OS X. It is an OKish workaroud.
```
cd packages/filesystem && yarn --ignore-scripts && ./node_modules/.bin/mocha -w --opts ../../configs/mocha.opts src/**/*.spec.ts
```

It does not work on Windows either from `GitBash` or `cygwin`. Obviously fails from `cmd.exe` too. (Did no try from PS).

From GitBash:
```
$ cd packages/filesystem && yarn --ignore-scripts && ./node_modules/.bin/mocha -w --opts ../../configs/mocha.opts src/**/*.spec.ts
yarn install v1.2.1
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.87s.

Error: The specified module could not be found.
\\?\C:\Users\kittaakos\dev\theia\node_modules\nsfw\build\Release\nsfw.node
    at Object.Module._extensions..node (module.js:653:18)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\kittaakos\dev\theia\node_modules\nsfw\lib\src\index.js:3:16)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\kittaakos\dev\theia\packages\filesystem\src\node\nsfw-watcher\nsfw-filesystem-watcher.ts:9:1)
    at Module._compile (module.js:624:30)
    at Module.m._compile (C:\Users\kittaakos\dev\theia\node_modules\ts-node\src\index.ts:392:23)
    at Module._extensions..js (module.js:635:10)
    at Object.require.extensions.(anonymous function) [as .ts] (C:\Users\kittaakos\dev\theia\node_modules\ts-node\src\index.ts:395:12)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\kittaakos\dev\theia\packages\filesystem\src\node\node-filesystem.spec.ts:20:1)
    at Module._compile (module.js:624:30)
    at Module.m._compile (C:\Users\kittaakos\dev\theia\node_modules\ts-node\src\index.ts:392:23)
    at Module._extensions..js (module.js:635:10)
    at Object.require.extensions.(anonymous function) [as .ts] (C:\Users\kittaakos\dev\theia\node_modules\ts-node\src\index.ts:395:12)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at C:\Users\kittaakos\dev\theia\node_modules\mocha\lib\mocha.js:231:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (C:\Users\kittaakos\dev\theia\node_modules\mocha\lib\mocha.js:228:14)
    at Mocha.run (C:\Users\kittaakos\dev\theia\node_modules\mocha\lib\mocha.js:514:10)
    at loadAndRun (C:\Users\kittaakos\dev\theia\node_modules\mocha\bin\_mocha:437:22)
    at Object.<anonymous> (C:\Users\kittaakos\dev\theia\node_modules\mocha\bin\_mocha:454:3)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Function.Module.runMain (module.js:665:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
```

### PR DESCRIPTION
**Pull request by kittaakos**
Closes #924.

Signed-off-by: Akos Kitta <kittaakos@gmail.com>

### CODE REVIEW

File: doc/Testing.md [hunk 1]
```diff
@@ -8,6 +8,20 @@ To run tests on theia run:
 
 This will run all CI enabled tests.
 
+If you want to run all tests for a particular Theia extension, execute the following command from the root:
+
+`npx run test @thiea/extension-name`
```
Comment (hexa00) [inline doc/Testing.md]: typo `s/thiea/theia`

File: doc/Testing.md [hunk 2]
```diff
@@ -8,6 +8,20 @@ To run tests on theia run:
...
+
+If you would like to enable watch mode for the tests, add the following npm script to the `package.json` of the desired Theia extension.
+
+```json
+  "test:watch": "theiaext test:watch"
+```
+
+After editing the `package.json`, execute `yarn` in the Theia root, then you can run the tests in watch mode with:
+
+`npx run test:watch @thiea/extension-name`
```
Comment (hexa00) [inline doc/Testing.md]: typo `s/thiea/theia`

File: doc/Testing.md (hunk 1 above)
Comment (kittaakos) [reply doc/Testing.md]: :( Thanks for the note.

**Comment by kittaakos** [comment]
> So this will compile and run the tests on file change ?

Yes. But it does not use `tsc` to recompile, but `ts-node`.

File: doc/Testing.md [hunk 3]
```diff
@@ -8,6 +8,20 @@ To run tests on theia run:
...
+
+`npx run test @theia/extension-name`
+
+If you would like to enable watch mode for the tests, add the following npm script to the `package.json` of the desired Theia extension.
+
+```json
+  "test:watch": "theiaext test:watch"
+```
+
+After editing the `package.json`, execute `yarn` in the Theia root, then you can run the tests in watch mode with:
```
Comment (akosyakov) [inline doc/Testing.md]: it is not required after changing `package.json`, calling yarn should be enough once after cloning the repo

Comment (kittaakos) [reply doc/Testing.md]: It was not the case, that's why I have [asked](https://github.com/theia-ide/theia/issues/924#issuecomment-348461744).

Comment (akosyakov) [reply doc/Testing.md]: I am sure that `theiaext` in binaries of all extensions after running yarn and adding a new npm scirpt does not change there theiext should be added as a binary or not. Maybe you did something like `git clean -dfx` and forget to run `yarn` again?

Comment (kittaakos) [reply doc/Testing.md]: Alright, I'll try it again before merging this PR.

**Pull request merged**